core/units.c

```diff
@@ -183,3 +186,49 @@ const struct units *get_units()
 {
 	return &prefs.units;
 }
+
+int parse_duration(const char *buffer, duration_t *time, bool dot_as_seperator)
+{
+	char *line = (char*)buffer;
+	size_t n = 0;
+	long values[3];
+	char separator;
+	while (n < sizeof(values)/sizeof(values[0])) {
+		values[n++] = strtol(line, &line, 10);
+		separator = *line;
+		line++;
```

Comment:
What if we hit the terminal `\0' here? Then line is now pointing past the buffer?